lib/libzfs_core/libzfs_core.c

```diff
@@ -645,6 +646,58 @@ lzc_send_resume(const char *snapname, const char *from, int fd,
 	    resumeoff, NULL));
 }
 
+struct sendargs {
+	int ioctlfd;
+	int inputfd;
+	int outputfd;
+};
+typedef struct sendargs sendargs_t;
+
+void *
+__do_send_output(void *voidargs)
+{
+	sendargs_t *args = (sendargs_t *)voidargs;
+	sigset_t sigs;
+	int buflen = 131072;
```

Comment:
I yanked the logic from libzfs into a new function in libzfs_core so it could be the same code called from both locations.

I find it pretty ugly to do, tbh, but we'll see if anyone else agrees with me...the alternative was duplicating the logic either inline or in a separate function in libzfs_core anyway, so this seemed least bad.